sda/processor.go

```diff
@@ -25,13 +116,16 @@ type ServiceProcessor struct {
 
 // NewServiceProcessor initializes your ServiceProcessor.
 func NewServiceProcessor(c *Context) *ServiceProcessor {
-	return &ServiceProcessor{
+	s := &ServiceProcessor{
 		functions: make(map[network.MessageTypeID]interface{}),
 		Context:   c,
 	}
+	return s
 }
 
 // RegisterMessage puts a new message in the message-handler
+// XXX More comments are needed as it's not clear whether RegisterMessage waits
```

Comment:
The question is more like "When" ^^ If you have any ideas how to clarify this function, please !